macaroonbakery/tests/test_utils.py

```diff
@@ -19,3 +23,14 @@ def test_cookie_expires_with_timezone(self):
         self.assertRaises(
             ValueError, cookie, 'http://example.com', 'test', 'value',
             expires=timestamp)
+
+
+class MacaroonToDictTest(TestCase):
+    def test_macaroon_to_dict(self):
+        m = pymacaroons.Macaroon(
+            key=b'rootkey', identifier=b'some id', location='here', version=2)
+        as_dict = bakery.macaroon_to_dict(m)
```

Comment:
it's preferred to just implement the dict interface and to then just cast it to a dict vs a named function for this. Is that an option? If I recall you'd just implement __iter__ on the object.

Reply:
Unfortunately, as discussed online, this isn't an option, as you can't just make arbitrary objects implement custom JSON encoding without changing the caller too.